x-pack/platform/plugins/shared/streams_app/public/components/stream_detail_overview/stream_detail_overview.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useMemo } from 'react';
+import { IngestStreamGetResponse, isWiredStreamDefinition } from '@kbn/streams-schema';
+import { ILM_LOCATOR_ID, IlmLocatorParams } from '@kbn/index-lifecycle-management-common-shared';
+
+import { computeInterval } from '@kbn/visualization-utils';
+import { useKibana } from '../../hooks/use_kibana';
+import { useStreamsAppFetch } from '../../hooks/use_streams_app_fetch';
+import { StreamsAppSearchBar } from '../streams_app_search_bar';
+import { getIndexPatterns } from '../../util/hierarchy_helpers';
+import { useDataStreamStats } from '../data_management/stream_detail_lifecycle/hooks/use_data_stream_stats';
+import { QuickLinks } from './quick_links';
+import { ChildStreamList } from './child_stream_list';
+import { StreamStatsPanel } from './components/stream_stats_panel';
+import { StreamChartPanel } from './components/stream_chart_panel';
+import { TabsPanel } from './components/tabs_panel';
+
+export function StreamDetailOverview({ definition }: { definition?: IngestStreamGetResponse }) {
+  const {
+    dependencies: {
+      start: {
+        data,
+        dataViews,
+        streams: { streamsRepositoryClient },
+        share,
+      },
+    },
+  } = useKibana();
+
+  const {
+    timeRange,
+    setTimeRange,
+    absoluteTimeRange: { start, end },
+  } = data.query.timefilter.timefilter.useTimefilter();
+
+  const indexPatterns = useMemo(() => {
+    return getIndexPatterns(definition?.stream);
+  }, [definition]);
+
+  const discoverLocator = useMemo(
+    () => share.url.locators.get('DISCOVER_APP_LOCATOR'),
+    [share.url.locators]
+  );
+
+  const bucketSize = useMemo(() => computeInterval(timeRange, data), [data, timeRange]);
```

Comment:
Lifecycle graph uses the [data-plugin](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_lifecycle/hooks/use_ingestion_rate.tsx#L45) logic to calculate interval.
Not sure how much it differs from this one but it would be nice to align

Reply:
The `computeInterval` is what's used by the ESQL histogram AFAICT, seems like a good standard. The one from the data plugin seems like it's used for the KQL discover bar and is more configurable, I don't think we need to lean into these here.

So if anything I would vote for using `computeInterval` everywhere.